pandas/_libs/tslibs/meson.build

```diff
@@ -0,0 +1,179 @@
+# TODO: can this be removed, I wish meson copied .pyx source to the build dir automatically
```

Comment:
We should be able to get rid of `__init__.py` file copying after https://github.com/cython/cython/pull/4906 is in a released Cython version.

For the `.pxd` files, are you copying them because they need to work with generated `.pyx` files? Or because you want to install them, or for some other reason?

I've been fighting with Cython quite a lot, but I've never found the need to copy `.pyx` files to the build dir - why does that help?

Reply:
The generated .pxi files are in the build directory since meson forces out-of-tree builds. Unfortunately, that means the Cython will not find them by default, since it only looks in the current directory where the .pyx files are. 

I'm trying to find an option for the ``cython`` command that tells it to look in other directories, but haven't found any luck there, so I'm hacking around it by copying and compiling the pyx files inside the build directory.

Reply:
Ah yes, that makes sense. Generated Cython code is a pain, we're trying to actively improve that in Cython and Meson. SciPy has a lot of generated Cython files; we use a Meson `generator` to automate dealing with copying files to the build dir: https://github.com/scipy/scipy/blob/8837756d06218936353895e6c30e5ca2b12c2e14/scipy/meson.build#L201-L211

Reply:
Do you think this is something that structured sources would fix? (mesonbuild/meson#9339)

Reply:
I'm not that confident to be honest; it's pretty complex and generated file objects are sometimes not treated the same as files specified as strings. Also, the conceptual issue is that it's baking in what I consider a design flaw in Cython; I'm more enthusiastic about fixing that. If with `--module-name` we get rid of needing to care about `__init__.py` files, and we can get Meson to understand what `.pxd` and `.pxi` files are and generate the appropriate `--include-dir` argument for them, then Cython is no longer structured.